### ISSUE TITLE
AddColumnSQL missing in  Data Dictionary for MySQL

### ISSUE THREAD

**Issue by ekariz** [ADOdb/ADOdb#1010]
### Description
AddColumnSQL function missing for mysql. I cannot add a new column / alter mysql table

### Environment
- ADOdb version: _Version number (or Git Commit/Ref)_
- Driver or Module:  mysql
- Database type and version:  mysql  Ver 8
- PHP version:  8.2
- Platform:  Linux (Ubuntu 20)

* [x] I have tested that the problem is reproducible in the [latest release](https://github.com/ADOdb/ADOdb/releases/latest) / **master** branch / **hotfix** branch  
  ⚠️ Keep only the applicable option(s), i.e. where you actually tested and remove the others)

### Steps to reproduce
Detailed, step-by-step instructions to reproduce the behavior, including:

$schema = new adoSchema($this->dbconn);
 
// Post Execution (default)
$schema->executeInline(false);
$schema->parseSchemaFile('schema.xml');
$schema->executeSchema();

if you add a new field in schema.xml , if you execute the datadictionary again , the new column is not added in the db table.

### Expected behavior
Adding new column in xml to mysql table

### Additional context 
This was working with older ADOdb versions

**Comment by dregad** [comment]
Can you please 
- provide a sample XML file (before/after) so we have a consistent reproducible test case, 
- describe how *this->dbconn* is initialized
- indicate which ADOdb version you're using, and in which "older ADOdb versions" this was previously working

**Comment by ekariz** [comment]
XML before running in db

![image](https://github.com/ADOdb/ADOdb/assets/5812903/f52435fd-cac4-4c08-bd61-118fed6f7b71)

Resulting SQL

![image](https://github.com/ADOdb/ADOdb/assets/5812903/146732fd-0e36-4893-8895-b18369fc025b)

Resulting mysql table
![image](https://github.com/ADOdb/ADOdb/assets/5812903/97a1a323-0ba8-4cf4-b5ee-0c04b1df64d2)

Added more columns in xml

![image](https://github.com/ADOdb/ADOdb/assets/5812903/da7e59b8-13e9-427f-a1c6-325247a9da00)

Resulting SQL AFTER adding column in XML

![image](https://github.com/ADOdb/ADOdb/assets/5812903/e1a346b0-0ba2-4e98-94cd-b39d37b24ec9)

MYSQL Table not modified to ADD new columns

![image](https://github.com/ADOdb/ADOdb/assets/5812903/f74031aa-ba6f-4a72-a201-475de49ffb95)


In Summary, i expect this
`ALTER TABLE tbl_test ADD COLUMN column2 VARCHAR(30) NOT NULL`

not this
` ALTER TABLE tbl_test MODIFY COLUMN column2 VARCHAR(30) NOT NULL`

**Comment by ekariz** [comment]
This is my adodb version
v5.22.5  2023-04-03

connection method
![image](https://github.com/ADOdb/ADOdb/assets/5812903/784cd83a-0c0d-454c-8181-4048b4b9b3f3)

**Comment by ekariz** [comment]
adodb_logsql errors
![image](https://github.com/ADOdb/ADOdb/assets/5812903/1cb1ad74-51cc-416d-837b-49c2106bbcc9)

**Comment by dregad** [comment]
Thanks for the details. I'll have a look as time allows. 

Note: For next time, better paste text (XML, SQL, PHP...) instead of images so I can copy-paste instead of re-type.

> This is my adodb version
> v5.22.5 2023-04-03

I suppose this is the version that has the problem ? You did not indicate which was the *older ADOdb version* that was working fine.

**Comment by ekariz** [comment]
First XML

<table name="tbl_test">
   <descr>test table</descr>
    <field name="id"            type="I" size="10"   ><NOTNULL/></field>
    <field name="column1"       type="C" size="30"   ><NOTNULL/></field>
   
  </table>

SQL
 
```
 CREATE TABLE tbl_test2 (
id    INTEGER(10) NOT NULL,
column1    VARCHAR(30) NOT NULL
)
```

Updated XML

```
<table name="tbl_test2">
  <descr>test table</descr>
  <field name="id"            type="I" size="10"   ><NOTNULL/></field>
  <field name="column1"       type="C" size="30"   ><NOTNULL/></field>
  <field name="column2"       type="C" size="30"   ><NOTNULL/></field>
  <field name="column3"       type="C" size="30"   ><NOTNULL/></field>

</table>
```
   
SQL

```
ALTER TABLE tbl_test2 MODIFY COLUMN id INTEGER(10) NOT NULL;
ALTER TABLE tbl_test2 MODIFY COLUMN column1 VARCHAR(30) NOT NULL;
ALTER TABLE tbl_test2 MODIFY COLUMN column2 VARCHAR(30) NOT NULL;
ALTER TABLE tbl_test2 MODIFY COLUMN column3 VARCHAR(30) NOT NULL;

```

**Comment by mnewnham** [comment]
It looks like a problem with the way that _**ADODB_DataDict::changeTableSQL()**_ is structured, when it iterates over the columns of an existing table, it doesn't distinguish between columns that already exist and those that don't. 

This looks like a problem caused by [This commit](https://github.com/ADOdb/ADOdb/commit/b5732c5a473a6d2bc92ac5d93683f4d7172c62f3)

**Comment by dregad** [comment]
Thanks for the analysis @mnewnham, that was on my to-do list for this week-end. Are you planning to work on a fix ?

**Comment by mnewnham** [comment]
@dregad I won't be able to get to it in the next 7 days

**Comment by ekariz** [comment]
Hi, just a follow up on this issue. Is a there a possible fix being worked  on ?

**Comment by dregad** [comment]
Hi @ekariz this fell off the radar, I have not looked at it since November last year. I'm definitely not working on it at all, and won't have time to do so in the short term, sorry.

**Comment by mnewnham** [comment]
I've looked at this again and you seem to have fixed this with 8d88183538ae08b86c30af5b0f3280c2accd71c3 @dregad

**Comment by dregad** [comment]
> I've looked at this again and you seem to have fixed this with [8d88183](https://github.com/ADOdb/ADOdb/commit/8d88183538ae08b86c30af5b0f3280c2accd71c3) @dregad

Indeed 😄 

That should have been committed in hotfix/5.22 branch though... I'll back-port it.

While retesting, I noticed that I forgot to initialize the *$fields_to_add* and *$fields_to_alter* arrays, resultsing in PHP warnings:
```
Notice: Undefined variable: fields_to_add in ./adodb-datadict.inc.php on line 1087
PHP Warning:  Invalid argument supplied for foreach() in ./adodb-datadict.inc.php on line 679
```

Also, I believe there is something wrong with the logic to drop columns when *$dropOldFlds* is true at the end of the changeTableSQL() method, as the loop refers to an undefined *$lines* array. Change was introduced by b5732c5a473a6d2bc92ac5d93683f4d7172c62f3. I'll open a separate issue to track this.

**Comment by ekariz** [comment]
Thanks @dregad  and @mnewnham
